homeassistant/components/filesize/sensor.py

```diff
@@ -52,48 +56,38 @@ def setup_platform(
 class Filesize(SensorEntity):
     """Encapsulates file size information."""
 
-    def __init__(self, path):
+    _attr_native_unit_of_measurement = DATA_MEGABYTES
+    _attr_icon = ICON
+
+    def __init__(self, path: str) -> None:
         """Initialize the data object."""
         self._path = path  # Need to check its a valid path
-        self._size = None
-        self._last_updated = None
-        self._name = path.split("/")[-1]
-        self._unit_of_measurement = DATA_MEGABYTES
+        self._size: int | None = None
+        self._last_updated: str | None = None
+        self._attr_name = path.split("/")[-1]
+        self._attr_unique_id = path
 
-    def update(self):
+    def update(self) -> None:
         """Update the sensor."""
         statinfo = os.stat(self._path)
         self._size = statinfo.st_size
         last_updated = datetime.datetime.fromtimestamp(statinfo.st_mtime)
         self._last_updated = last_updated.isoformat()
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
+    def native_value(self) -> float | None:
         """Return the size of the file in MB."""
         decimals = 2
-        state_mb = round(self._size / 1e6, decimals)
-        return state_mb
+        if self._size:
+            state_mb = round(self._size / 1e6, decimals)
+            return state_mb
```

Comment:
Why not return this instead of storing it first?
```suggestion
            return round(self._size / 1e6, decimals)
```